cmake/add_enclave.cmake

```
@@ -121,15 +122,22 @@ endfunction()
 
 function(add_enclave_sgx)
   set(options CXX)
-  set(oneValueArgs TARGET CONFIG KEY SIGNING_ENGINE ENGINE_LOAD_PATH ENGINE_KEY_ID)
+  set(oneValueArgs TARGET CONFIG KEY SIGNING_ENGINE ENGINE_LOAD_PATH ENGINE_KEY_ID ADD_LVI_MITIGATION)
   set(multiValueArgs SOURCES)
   cmake_parse_arguments(ENCLAVE
     "${options}"
     "${oneValueArgs}"
     "${multiValueArgs}"
     ${ARGN})
 
-  add_enclave_executable(${ENCLAVE_TARGET} ${ENCLAVE_SOURCES})
+  add_executable(${ENCLAVE_TARGET} ${ENCLAVE_SOURCES})
+  # Add an enclave with LVI mitigation when the `ADD_LVI_MITIGATION` argument (with `ENCLAVE` prefix)
+  # is set or the `LVI_MITIGATION_SKIP_TESTS` global variable is not set.
```

Given the confusion in our conversation on the review thread, I suggest explaining the semantics rather that rephrasing the boolean logic that can be read from the code:
```suggestion
  # Add an enclave with LVI mitigation if LVI_MITIGATION is globally configured.
  #
  # If the LVI_MITIGATION_SKIP_TESTS global variable is set, then it takes
  # precedence and suppress the addition of LVI mitigated binaries (which are
  # primarily test binaries in the OE SDK). This variable also skips adding ctests
  # for the LVI mitigated binaries in add_enclave_test.cmake.
  #
  # The ADD_LVI_MITIGATION argument to add_enclave() can override LVI_MITIGATION_SKIP_TESTS
  # on a per enclave basis. This parameter has no effect if either LVI_MITIGATION or
  # LVI_MITIGATION_SKIP_TESTS is not specified.
  # It only re-enables the additional LVI-mitigated build of the specified enclave.
  # It does not enable the additional ctest against the LVI-mitigated version of
  # the enclave.
```